app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -82,19 +81,19 @@ def not_found(options = nil)
     false
   end
 
-  def registration_url
-    uri = if params[:url].present?
-            URI.join(params[:url], '/register')
-          else
-            URI(register_url)
-          end
+  def url
+    uri = URI(params[:url] || root_url)
 
     return uri if uri.scheme && uri.host
 
     msg = N_('URL in :url parameter is missing a scheme, please set http:// or https://')
     fail Foreman::Exception.new(msg)
   end
 
+  def context_urls
+    { url: url, registration_url: URI.join(url, 'register') }
```

Comment:
I think `url` is still ambiguous and also something we previously didn't have. Why is it needed?